packages/sirius-web/backend/sirius-web-application/src/main/java/org/eclipse/sirius/web/application/representation/services/RepresentationSearchService.java

```diff
@@ -42,21 +50,28 @@ public class RepresentationSearchService implements IRepresentationSearchService
 
     private final Logger logger = LoggerFactory.getLogger(RepresentationSearchService.class);
 
-    public RepresentationSearchService(IRepresentationDataSearchService representationDataSearchService, ObjectMapper objectMapper) {
-        this.representationDataSearchService = Objects.requireNonNull(representationDataSearchService);
+    private final List<IRepresentationMigrationParticipant> migrationParticipants;
+
+    private final IRepresentationDataUpdateService representationDataUpdateService;
```

Comment:
Don't mix services like that, they are supposed to be independent. The search service is not supposed to reuse the update service. The search service is supposed to be 100% side effect free with no modification of the database while the update service is doing the exact opposite